src/switchcraft_advanced/utils/winget.py

```
@@ -174,15 +122,20 @@ def search_packages(self, query: str) -> List[Dict[str, str]]:
             proc = subprocess.run(cmd, capture_output=True, text=True, encoding="utf-8", errors="ignore", startupinfo=startupinfo)
 
-            if proc.returncode != 0 and proc.returncode != 1: # 0=Success, 1=No results
+            # If nothing found, winget often returns 1 and prints "No package found..."
+            if "No package found" in proc.stdout:
                 return []
 
+            if proc.returncode != 0 and proc.returncode != 1: # 0=Success, 1=No results
+                logger.error(f"Winget CLI error {proc.returncode}: {proc.stderr or proc.stdout}")
+                # We can return an error dict, but type hint says List
+                # Or we raise an exception to be caught by UI
+                raise RuntimeError(f"Winget Error {proc.returncode}: {proc.stdout}")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)